client/my-sites/post-selector/selector.jsx

```diff
@@ -340,48 +358,50 @@ const PostSelectorPosts = React.createClass( {
 	},
 
 	render() {
-		const rowCount = this.getRowCount();
+		const { className, siteId, query } = this.props;
+		const { requestedPages, searchTerm } = this.state;
 		const isCompact = this.isCompact();
 		const isTypeLabelsVisible = this.isTypeLabelsVisible();
-		const showSearch = this.state.searchTerm.length > 0 || ! isCompact;
-		const classes = classNames( 'post-selector', this.props.className, {
-			'is-loading': this.props.loading,
+		const showSearch = searchTerm.length > 0 || ! isCompact;
+		const classes = classNames( 'post-selector', className, {
 			'is-compact': isCompact,
 			'is-type-labels-visible': isTypeLabelsVisible
 		} );
 
 		return (
-			<div ref={ this.setSelectorRef } className={ classes }>
-				<QueryPosts
-					siteId={ this.props.siteId }
-					query={ this.props.query } />
+			<div className={ classes }>
+				{ requestedPages.map( ( page ) => (
+					<QueryPosts
+						key={ `page-${ page }` }
+						siteId={ siteId }
+						query={ { ...query, page } } />
+				) ) }
 				{ isTypeLabelsVisible && (
-					<QueryPostTypes siteId={ this.props.siteId } />
+					<QueryPostTypes siteId={ siteId } />
 				) }
 				{ showSearch && (
 					<Search
-						searchTerm={ this.state.searchTerm }
+						searchTerm={ searchTerm }
 						onSearch={ this.onSearch } />
 				) }
-				<InfiniteLoader
-					ref="loader"
-					isRowLoaded={ this.isRowLoaded }
-					loadMoreRows={ this.maybeFetchNextPage }
-					rowCount={ rowCount }>
-					{ ( { onRowsRendered, registerChild } ) => (
-						<VirtualScroll
-							ref={ registerChild }
-							width={ this.getResultsWidth() }
-							height={ isCompact ? this.getCompactContainerHeight() : 300 }
-							onRowsRendered={ onRowsRendered }
-							noRowsRenderer={ this.renderEmptyContent }
-							rowCount={ rowCount }
-							estimatedRowSize={ ITEM_HEIGHT }
-							rowHeight={ this.getRowHeight }
-							rowRenderer={ this.renderRow }
-							className="post-selector__results" />
-					) }
-				</InfiniteLoader>
+				<div className="post-selector__results">
+					<AutoSizer
+						key={ JSON.stringify( query ) }
```

Comment:
>Just curious why you need to set a key here?

As in #6427, there's a few gotchas with `ReactVirtualized`, specifically when the row count doesn't change. The idea with adding `key` here is to address cases such as where a query changes but the resulting `postsHierarchy` results in the `rowCount` remaining the same (in fact, this might be an issue in production currently).